GooglePlayInstant/Editor/PlayInstantBuilder.cs

```diff
@@ -212,6 +214,13 @@ public static bool BuildAndSign(BuildPlayerOptions buildPlayerOptions)
         /// <returns>True if the build succeeded, false if it failed or was cancelled.</returns>
         public static bool Build(BuildPlayerOptions buildPlayerOptions)
         {
+            var checkInstantManifestResult = AndroidManifestHelper.GetAndroidManifestUpdater().CheckInstantManifest();
```

Comment:
If we're already checking the version here, should we instead call SwitchToInstant here? I'm thinking about cases where we update the plugin with some new AndroidManifest requirement, but their manifest doesn't get rebuilt because their project is already in instant mode. It would also make the Legacy codepath more in line with the IPostGenerateGradleAndroidProject method.

Reply:
Good question. I also considered calling SwitchToInstant here.

The really nice thing about IPostGenerateGradleAndroidProject on 2018+ is that we apply our changes to the fully merged AndroidManifest.xml right before the build. As far as I know, it always works.

The pre-2018 approach where we apply changes to one of possibly many manifests pre-merge is more brittle, and I've heard several people from other Google Unity projects say as much. I expect that there may be cases where after the developer saves the manifest in Instant mode, that they have to go in and manually modify the manifest for some reason or another.

My thinking here was to perform the least invasive change possible - just update the plugin version. Also, I only save the file if it needs an update, which avoids file I/O and potentially causing Unity to need to refresh assets.

In short - this approach is a bit more code, but seems less risky to me.

Reply:
Yes, that's fair. We could check the file for correctness and at least prompt a warning if anything seems amiss. I will plan to save that as future work.